File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/repository/CardRepository.java
package com.lattis.ellipse.domain.repository;

import com.lattis.ellipse.data.network.model.response.BasicResponse;
import com.lattis.ellipse.data.network.model.response.card.AddCardResponse;
import com.lattis.ellipse.data.network.model.response.card.SetUpIntentDataResponse;
import com.lattis.ellipse.data.network.model.response.card.SetUpIntentResponse;
import com.lattis.ellipse.domain.model.Card;

import org.json.JSONObject;

import java.util.List;

import io.reactivex.Observable;

/**
 * Created by ssd3 on 7/26/17.
 */

public interface CardRepository {
    Observable<AddCardResponse> addCard(String cc_no, int exp_month, int exp_year, String cvc, JSONObject intent);
    Observable<List<Card>> getCard();

    Observable<BasicResponse> deleteCard(int cardId);

    Observable<BasicResponse> updateCard(int cardId);

    Observable<SetUpIntentResponse> getSetUpIntent();
}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/repository/CardRepository.kt
package com.lattis.domain.repository

import com.lattis.domain.models.SetUpIntent
import com.lattis.domain.models.Card
import com.lattis.domain.models.MPPublicKey
import io.reactivex.rxjava3.core.Observable
import org.json.JSONObject

interface CardRepository {
    fun getCard(): Observable<List<Card>>

    fun deleteCard(cardId: Int): Observable<Boolean>

    fun updateCard(cardId: Int): Observable<Boolean>

    fun getSetUpIntent(): Observable<SetUpIntent>

    fun addCard(
        cc_no: String,
        exp_month: Int,
        exp_year: Int,
        cvc: String,
        intent: JSONObject
    ): Observable<Boolean>

    fun getMPPublicKey(
        fleet_id: Int
    ): Observable<MPPublicKey>

    fun addMPCard(
        token:String,
        fleet_id: Int?
    ):Observable<Boolean>

    fun updateCardExpiration(cardId: String,exp_month: Int, exp_year: Int): Observable<Boolean>

}